lib/Conversion/TTIRToTTMetal/TTIRToTTMetal.cpp

```diff
@@ -741,36 +750,137 @@ class TTIRToTTMetalDispatchRewriter : public OpRewritePattern<ttir::GenericOp> {
       llvm_unreachable("Unhandled unary op compute conversion.");
     }
+
+    builder.create<ttkernel::TileRegsCommitOp>(arithOrMathOp.getLoc());
+
+    builder.create<ttkernel::TileRegsWaitOp>(arithOrMathOp.getLoc());
+
+    // Copy tile from DST at dstTileIndex to outCB at outCBTileIndex.
+    // outCBTileIndex increments as loops iterate, thus placing one result tile
+    // after another in outCB.
+    builder.create<ttkernel::PackTileOp>(arithOrMathOp.getLoc(), dstTileIndex,
+                                         outCB, outCBTileIndex);
+
+    // PACK releases lock on DST.
+    builder.create<ttkernel::TileRegsReleaseOp>(arithOrMathOp.getLoc());
   }
 
   void convertComputeBinaryOp(Operation &arithOrMathOp,
                               ArrayRef<BlockArgument> cbOperands,
                               ArrayRef<BlockArgument> iterators,
                               SmallVector<unsigned> blockArgIteratorMapping,
-                              Value dstTileIndex, OpBuilder &builder) const {
+                              OpBuilder &builder) const {
     assert(cbOperands.size() == 3 &&
            "Expected two input and one output CB for binary op.");
 
     auto inCB0TileIndex = iterators[blockArgIteratorMapping[0]];
     auto inCB0 = cbOperands[0];
     auto inCB1TileIndex = iterators[blockArgIteratorMapping[1]];
     auto inCB1 = cbOperands[1];
+    auto outCB = cbOperands[2];
 
     // Perform computation C = A (*) B on tile A from inCB0 and tile B from
     // inCB1 and store the result C in DST register on dstTileIndex.
+
     if (mlir::isa<arith::AddFOp>(arithOrMathOp)) {
+      Value dstIndex = i32(0, builder);
+      builder.create<ttkernel::TileRegsAcquireOp>(arithOrMathOp.getLoc());
       builder.create<ttkernel::AddTilesOp>(arithOrMathOp.getLoc(), inCB0, inCB1,
                                            inCB0TileIndex, inCB1TileIndex,
-                                           dstTileIndex);
+                                           dstIndex);
+      builder.create<ttkernel::TileRegsCommitOp>(arithOrMathOp.getLoc());
+      builder.create<ttkernel::TileRegsWaitOp>(arithOrMathOp.getLoc());
+      builder.create<ttkernel::PackTileOp>(
+          arithOrMathOp.getLoc(), dstIndex, outCB,
+          iterators[blockArgIteratorMapping[2]]);
+      builder.create<ttkernel::TileRegsReleaseOp>(arithOrMathOp.getLoc());
     } else if (mlir::isa<arith::MulFOp>(arithOrMathOp)) {
-      builder.create<ttkernel::MulTilesOp>(arithOrMathOp.getLoc(), inCB0, inCB1,
-                                           inCB0TileIndex, inCB1TileIndex,
-                                           dstTileIndex);
+      commonComputeMulOp(arithOrMathOp, cbOperands, iterators,
+                         blockArgIteratorMapping, builder);
+    } else if (mlir::isa<arith::DivFOp>(arithOrMathOp)) {
+
+      SmallVector<std::int64_t> operandIndicesRecip;
+      // For DIV, input 1 is going through reciprocal.
+      operandIndicesRecip.push_back(1);
```

Comment:
What's the use of this if you know you are going to use `inCB1` in `commonComputeRecipOp` anyway?